### PR TITLE
Fix a couple of Jawn bugs.

### LINKED ISSUE #119: Inconsistencies in parsing failure location reporting
Just making a note of this so I don't forget about it.

Given some invalid JSON like this:
```scala
import jawn.{ AsyncParser, NullFacade, Parser }
import java.io.ByteArrayInputStream, java.nio.channels.Channels

val json = "[1, 2,\nx3]"
val ch = Channels.newChannel(new ByteArrayInputStream(json.getBytes))
```
Different Jawn parser implementations give three different locations for the failure:
```scala
scala> println(Parser.parseFromString(json)(NullFacade))
Failure(jawn.ParseException: expected json value got x (line 2, column 8))

scala> println(Parser.parseFromChannel(ch)(NullFacade))
Failure(jawn.ParseException: expected json value got x (line 2, column 2))

scala> println(Parser.async(jawn.AsyncParser.UnwrapArray)(NullFacade).absorb(json)(NullFacade))
Left(jawn.ParseException: expected json value got x (line 2, column 1))
```
The async parser is the only correct one.

### LINKED ISSUE #122: Invalid Unicode escapes are accepted
I just ran Jawn 0.13.0 through the [Parsing JSON is a Minefield](http://seriot.ch/json/parsing.html) tests, and the one failure is for `n_string_invalid_unicode_escape.json`, where `["\uqqqq"]` is accepted even though it should not be:

```scala
scala> val input = "[\"\\uqqqq\"]"
input: String = ["\uqqqq"]

scala> jawn.ast.JParser.parseFromString(input)
res0: scala.util.Try[jawn.ast.JValue] = Success(["\u0000"])
```

### PR DESCRIPTION
**Pull request by non**
This commit fixes three issues:

 - SOE when parsing a huge amount of whitespace
 - Fix situations where broken unicode escapes are accepted
 - Fix a test bug where the async parser will accept empty strings

Fixes #119
Fixes #122

### CODE REVIEW

**Comment by non** [comment]
cc @travisbrown

**Comment by non** [comment]
Ugh dammit, apparently there's no way to use either with a for-comprehension that won't generate a warning or error across 2.11 - 2.13 😬

I'll refactor it as a pattern-match.

File: parser/src/main/scala/jawn/Parser.scala [hunk 1]
```diff
@@ -89,7 +89,8 @@ abstract class Parser[J] {
   protected[this] def column(i: Int): Int
 
   protected[this] final val HexChars: Array[Int] = {
```
Comment (travisbrown) [inline parser/src/main/scala/jawn/Parser.scala]: Also not a big deal, and not related to this PR, but is there a reason `HexChars` is an instance val? It seems unnecessary to instantiate this array for every string parsed.

**Comment by travisbrown** [comment]
:+1: , thanks!

File: parser/src/main/scala/jawn/Parser.scala [hunk 2]
```diff
@@ -89,7 +89,8 @@ abstract class Parser[J] {
...
-    val arr = new Array[Int](128)
+    //val arr = new Array[Int](128)
```
[the remark this answers is not in the document]
Comment (dwijnand) [reply parser/src/main/scala/jawn/Parser.scala]:
```suggestion
```

😄

[the remark this answers is not in the document]
Comment (non) [reply parser/src/main/scala/jawn/Parser.scala]: Good catch. Removed in newest commit.

File: parser/src/main/scala/jawn/Parser.scala (hunk 1 above)
Comment (non) [reply parser/src/main/scala/jawn/Parser.scala]: Sure! I moved all instance vals into the `Parser` companion except for `utf8` which is used in many of the subclasses. (I could move that one too but it seemed OK to leave it.)

**Comment by travisbrown** [comment]
:+1:

**Pull request merged**